driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -292,6 +300,121 @@ static GeoNearSearchOperator near(final Point origin, final Number pivot, final
+        return in(notNull("path", path), concat(notNull("value", value), values));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The uuid value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final UUID value, final UUID... values) {
+        List<BsonBinary> bsonValues = new ArrayList<>();
+        bsonValues.add(new BsonBinary(value));
+        if (values != null) {
+            for (UUID uuid : values) {
+                bsonValues.add(new BsonBinary(uuid));
+            }
+        }
+        return in(notNull("path", path), notNull("value", bsonValues));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The string value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final String value, final String... values) {
+        return in(notNull("path", path), concat(notNull("value", value), values));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that searches for an array of values at the given path and returns documents where the value of
+     * the field equals any value in the specified array.
+     *
+     * @param path The indexed field to be searched.
+     * @param values The non-empty values to search for. Value can be either a single value or an array of values of only one of the supported BSON types and can't be a mix of different types.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static <T> InSearchOperator in(final FieldSearchPath path, final Iterable<? extends T> values) {
+        Iterator<T> valueIterator = (Iterator<T>) notNull("values", values).iterator();
+        isTrueArgument("values must not be empty", valueIterator.hasNext());
+        T firstValue = valueIterator.next();
+
+        Iterator<T> typeIterator = (Iterator<T>) notNull("values", values).iterator();
+        while (typeIterator.hasNext()) {
+            Object element = typeIterator.next();
+            isTrueArgument("values must be of same type", firstValue.getClass().isInstance(element));
+        }
```

Comment:
I don't think we should iterate `values` here to check that they are of the same type as `firstValue`, because [_"Where possible, depend on server to return errors
"_](https://github.com/mongodb/specifications/blob/master/source/driver-mantras.md#where-possible-depend-on-server-to-return-errors).

If we see the the search building API currently violating that mantra somewhere else, we may add one more item to re-visit to [JAVA-5752: Re-visit some aspects of the Atlas search API design before moving it out of beta
](https://jira.mongodb.org/browse/JAVA-5752).